nimbus-ui/nimbusui/src/app/components/platform/form/elements/calendar.component.ts

```diff
@@ -59,20 +60,36 @@ export const CUSTOM_INPUT_CONTROL_VALUE_ACCESSOR: any = {
             [yearNavigator]="element.config?.uiStyles?.attributes?.yearNavigator"
             [readonlyInput]="element.config?.uiStyles?.attributes?.readonlyInput"
             [yearRange]="element.config?.uiStyles?.attributes?.yearRange"
+            [maxDate]="maxDate"
+            [minDate]="minDate"
             [disabled]="disabled">
         </p-calendar>
    `
 })
 export class Calendar extends BaseControl<Date> {
 
+    private minDate : Date;
+    private maxDate : Date;
     @ViewChild(NgModel) model: NgModel;
 
     constructor(wcs: WebContentSvc, controlService: ControlSubscribers, cd:ChangeDetectorRef) {
         super(controlService,wcs,cd);
     }
 
-    // ngOnInit(){
- 
-    // }
+    ngOnInit() {
+        this.applyDateConstraint()
+    }
+
+    private applyDateConstraint() {
+        let pastConstraint = this.getConstraint(ValidationConstraint._past.value);
+        let futureConstraint = this.getConstraint(ValidationConstraint._future.value);
+
+        if (pastConstraint) {
+            this.maxDate = new Date();
+        } else if (futureConstraint) {
```

Comment:
Consider moving this to separate individual if conditions, since if the config is given:
```java
@Past
@Future
@Calendar
private LocalDate dob;
```

Only the `@Past` would take effect.  Although the scenario is not valid, the UI framework should allow this as a configuration mishap.

Otherwise, if this scenario does not make sense we should consider throwing an error.